state/factory/statedb.go

```diff
@@ -251,8 +251,31 @@ func (sdb *stateDB) newWorkingSet(ctx context.Context, height uint64) (*workingS
 			flusher.KVStoreWithBuffer().MustDelete(ns, key)
 			return nil
 		},
-		statesFunc: func(opts ...protocol.StateOption) (uint64, state.Iterator, error) {
-			return sdb.States(opts...)
+		statesFunc: func(ns string, keys [][]byte) ([][]byte, error) {
+			if keys == nil {
```

Comment:
add a comment like
// no Keys set, default to get all key-values in the namespace